src/pipe.ts

```diff
@@ -76,7 +76,7 @@ export function pipe(
   const lazyOps = operations.map(op => {
     const { lazy, lazyArgs } = op as LazyOp;
     if (lazy) {
-      const fn: any = lazy(...lazyArgs);
+      const fn: any = lazy(...(lazyArgs ?? []));
```

Comment:
why was this change made ?

Reply:
> why was this change made ?

Butt protection. If lazyArgs is not provided, the destructuration of `undefined` would go :boom:.

Reply:
types make sure that you only pass functions into a pipe, not something that can be undefined / null. since this PR is about changing the test framework, I'd revert that change please.

Reply:
> types make sure that you only pass functions into a pipe, not something that can be undefined / null.

Assuming the end-user is using TypeScript, and didn't throw an any somewhere. I was thinking along the lines of "be strict in what you emit, lax in what you accept". But you are right, it's not related to the change of testing framework. I'll remove the line presently!

Reply:
Okay, this is funky. I removed that protection, and the code complains that lazyArgs is undefined. Okay, fair, that's probably why I put the code there in the first place. But it works on master. So I went back there and did what expert programmers do: added print statements. The result: https://github.com/remeda/remeda/compare/master...yanick:undefined-yet-spreadable?expand=1  

And... the log says that lazyArgs for `first` is undefined, and yet the next line that spread it out doesn't go boom. Even better, if I change the log to be `console.log('!', lazyArgs, [ ...lazyArgs ] )`, that one goes boom.

I suspect it's something I changed in the tsconfig? I'll bisect the changes and find out.

Reply:
> I suspect it's something I changed in the tsconfig? I'll bisect the changes and find out.

I think? I have an idea what's going on. `tsc` turns `lazy(...lazyArgs)` into `lazy.apply(void 0, lazyArgs)`, which is totally okay with `lazyArgs` being undefined. But `vitest` doesn't use `tsc` and mostly just strip away the type annotations, so the spread remains a spread.

So now I have to see why `first` in that test doesn't have a lazyArgs...

Reply:
I think I got it. Basically, calling `first` instead of `first()` in the pipe only work if we explicitly define the lazyArgs to be [].

Reply:
okay I mean we can keep the fallback, even it's only to make vite happy 🤷 . But please use `||` as nullish colesce will likely be transpiled to something larger

Reply:
Will do in a few seconds! And you got me curious: the `??` would be transpiled into 

```
// lazyArgs ?? [] turns into --v
(lazyArgs !== null && lazyArgs !== void 0 ? lazyArgs : [])
```